File: 백준/Silver/9012. 괄호/괄호.c
#include <stdio.h>
#include <string.h>

int main(void) {
	int t;
	scanf("%d", &t);
	char stack[51];
	int left = 0, right = 0;
	while (t > 0) {
		scanf("%s", stack);
		for (int i = 0; i < strlen(stack); i++) {
			if (stack[i] == '(') left++;
			else if (stack[i] == ')') right++;

			if (right > left) break;
		}
		if (left != right) printf("NO\n");
		else printf("YES\n");
		left = 0, right = 0;
		t--;
	}

	return 0;
}

File: 백준/Silver/9012. 괄호/괄호.py
n  = int(input())
stack = []
left = 0
right = 0
while(n > 0):
    ch = list(input())
    for i in ch:
        if (i == '('): left += 1
        elif (i == ')'): right += 1
        
        if(right > left): break
    if(left != right): print("NO")
    else: print("YES")
    left = 0
    right = 0    
    n -= 1